test/test_list_students_take_mod.py

```diff
@@ -28,19 +28,22 @@ def test_list_students_take_module_one_focus(self):
             to take a certain mod in a specified semester,
             where these students only have one focus area.
         '''
+        # Add some modules and dummy students
+        model.add_student('dummyYr1A', 1)
+        model.add_student_focus_area('dummyYr1A', 'Database Systems', None)
+        model.add_student_plan('dummyYr1A', True, 'CS1010', 'AY 16/17 Sem 1')
 
         list_of_students_take_mod = \
             model.get_list_students_take_module('CS1010', 'AY 16/17 Sem 1')
 
-        required_list = [['D1000002A', 1, 'Database Systems', '-'],
-                         ['D1000000A', 1, 'Artificial Intelligence', '-'],
-                         ['D1000001A', 1, 'Computer Graphics and Games', '-'],
-                         ['D5000000A', 1, 'Software Engineering', '-'],
-                         ['D5000001A', 2, 'Software Engineering', '-']
-                        ]
+        required_entry = ['dummyYr1A', 1, 'Database Systems', '-']
 
-        assert_equal(len(list_of_students_take_mod), len(required_list))
-        assert_equal(sorted(list_of_students_take_mod), sorted(required_list))
+        # Clean up database
+        model.delete_all_plans_of_student('dummyYr1A')
+        model.delete_student('dummyYr1A')
+        model.delete_student_focus_area('dummyYr1A')
```

Comment:
by right, you should delete_student_focus_area first before delete_student. Pls change this and add in a foreign key reference to the schema such that takesFocusArea's attribute nusnetId references foreign key from student's nusnetId